hkmc2/shared/src/test/mlscript/lifter/ModulesObjects.mls

```diff
@@ -286,3 +286,89 @@ fun foo(x) =
 
 foo(123)
 //│ = [true, [true, false], true]
+
+// NOTE: This problem have a related test in codegen/Modules.mls with comment "report initialization problem or make it work"
+// Maybe we can codegen A10 = A9; at the beginning of the static block?
+:sjs
+:fixme
+module A with
+  fun g = 1
+  fun f = () => g
+  f()
+//│ JS (unsanitized):
+//│ let A10, lambda;
+//│ lambda = (undefined, function () {
+//│   return A10.g
+//│ });
+//│ A10 = class A9 {
+//│   static {
+//│     A9.f()
+//│   }
+//│   static get g() {
+//│     return 1;
+//│   } 
+//│   static get f() {
+//│     return lambda;
+//│   }
+//│   static toString() { return "A"; }
+//│ };
+//│ ═══[RUNTIME ERROR] TypeError: Cannot read properties of undefined (reading 'g')
```

Comment:
I wonder if adding a simple fix for this would be great: i.e.
```js
class A9 {
  static {
    A10 = A9;
    A9.f()
    ...
```

Reply:
I've implemented this on the latest commit. Let me know if this solution is fine or not.

Reply:
Pls see https://github.com/hkust-taco/mlscript/pull/296/files#r2029701237